### PR TITLE
get patient appointments

### PR DESCRIPTION
**Pull request by g-vega-cl**
purpose:
 - Create `/:patientID/appointments` route in `patient.api` that returns a patient's relevant appointments in order.

Trello:
- https://trello.com/c/aqqEQwet/80-appointments-coaches-can-see-a-table-with-all-their-upcoming-appointments-sorted-soonest-to-not-soon

### CODE REVIEW

File: src/routes/patient.integration.test.ts [hunk 1]
```diff
@@ -69,6 +70,65 @@ if (process.env.NODE_ENV === 'development') {
...
+        const appointment = res.body[i];
+        if (new Date(appointment.scheduledFor) < lastAppointmentDate) {
+          isInOrder = false;
+          break;
+        }
+        if (!lastAppointmentDate) {
+          lastAppointmentDate = appointment.scheduledFor;
+        }
+      }
+      expect(isInOrder).toBeTruthy();
```
Comment (rweald) [inline src/routes/patient.integration.test.ts:122]: I'd probably make this test a lot easier to reason about by just expecting very explicit order in a traditional "equals" clause. 

It took me quite a bit of reasoning to understand what you were testing, when in reality we know from the creation of the text data, exactly what order items should be returned, and therefore can test that explicitly.

Comment (rweald) [reply src/routes/patient.integration.test.ts:122]: Something more like: 
```
expect(res.body).toEqual([someItem1, someItem2, someItem3])
```

Comment (g-vega-cl) [reply src/routes/patient.integration.test.ts:122]: I tried using `expect(res.body).toEqual([appointment1, appointment2, appointment3])` but the issue is that superset returns a string (well, a JSON), and appointments returns Date. So I did some parsing before the `toBe()`.

Comment (rweald) [reply src/routes/patient.integration.test.ts]: Ahh yeah, I think doing a bit of coercion is fine. Still much easier to read as an explicit ordering like you have now 👍

**Review by rweald — APPROVED** [review]

**Comment by rweald** [comment]
Let's squash merge this one since it has that extra test cleanup commit that should be part of the single final commit on master.

**Pull request merged**